lid.cpp

```diff
@@ -48,6 +37,89 @@ void Lid::validate(std::string filePath)
     return;
 }
 
+void Lid::assembleCodeUpdateImage()
+{
+    info("InbandCodeUpdate: assembleCodeUpdateImage\n");
+    utils::subscribeToSystemdSignals(bus);
+    auto method = bus.new_method_call(SYSTEMD_BUSNAME, SYSTEMD_PATH,
+                                      SYSTEMD_INTERFACE, "StartUnit");
+    method.append("assemble-lids.service", "fail");
+    bus.call_noreply(method);
+}
+
+void Lid::unitStateChange(sdbusplus::message_t& msg)
+{
+    uint32_t newStateID{};
+    sdbusplus::message::object_path newStateObjPath;
+    std::string newStateUnit{};
+    std::string newStateResult{};
+
+    // Read the msg and populate each variable
+    msg.read(newStateID, newStateObjPath, newStateUnit, newStateResult);
+
+    auto assembleLidServiceFile = "assemble-lids.service";
+
+    if (newStateUnit == assembleLidServiceFile)
+    {
+        info(
+            "InbandCodeUpdate: Assemble Lid Service file Return value - {SERVICERESULT}",
+            "SERVICERESULT", newStateResult);
+        // unsubscribe to systemd signals
+        utils::unsubscribeFromSystemdSignals(bus);
+
+        if (newStateResult == "done")
+        {
+            info(
+                "InbandCodeUpdate: Assemble Code update image is completed successfully.");
+        }
+        else
+        {
+            // Set Activation value to Failed
```

Comment:
Strictly speaking, the item updater will set the Activation value to Invalid instead of Failed.